### PR TITLE
chore: remove deprecated `app.runningUnderRosettaTranslation()`

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Follow-up to #39897

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes
Notes: Removed deprecated `app.runningUnderRosettaTranslation()`

### CODE REVIEW

**Review by zcbenz — APPROVED** [review]

File: patches/node/tls_ensure_tls_sockets_are_closed_if_the_underlying_wrap_closes.patch [hunk 1]
```diff
@@ -17,10 +17,10 @@ Reviewed-By: Matteo Collina <matteo.collina@gmail.com>
 Reviewed-By: Debadree Chatterjee <debadree333@gmail.com>
 
 diff --git a/lib/_tls_wrap.js b/lib/_tls_wrap.js
-index 1eff3b3fba8a05d5fade43c6cb00b6621daa8c3d..5bed23e1355e5e5a1965f15ca270fb372f751d66 100644
+index 84599102be4b74ff66af63c36c232f4656a1f406..bb0535558a8999034a99d0e7e99507ba10eca2eb 100644
 --- a/lib/_tls_wrap.js
 +++ b/lib/_tls_wrap.js
-@@ -629,6 +629,9 @@ TLSSocket.prototype._wrapHandle = function(wrap, handle) {
+@@ -661,6 +661,9 @@ TLSSocket.prototype._wrapHandle = function(wrap, handle, wrapHasActiveWriteFromP
```
Comment (RaisinTen) [inline patches/node/tls_ensure_tls_sockets_are_closed_if_the_underlying_wrap_closes.patch:23]: Unrelated change?

**Review by codebytere — APPROVED** [review]

File: patches/node/tls_ensure_tls_sockets_are_closed_if_the_underlying_wrap_closes.patch (hunk 1 above)
Comment (codebytere) [reply patches/node/tls_ensure_tls_sockets_are_closed_if_the_underlying_wrap_closes.patch]: rebased and it's fixed

**Review by codebytere — APPROVED** [review]
oops accidentally re-requested myself

**Review by RaisinTen — APPROVED** [review]
lgtm

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Comment by miniak** [comment]
@codebytere can you please approve / merge?

**Review by codebytere — APPROVED** [review]
API LGTM
